### PR TITLE
Enhance/#6584 - Prevent flicker on AdSense, Analytics and Tag Manager setup screens

### PR DESCRIPTION
**Pull request by hussain-t**
## Summary

Addresses issue:

- #6584 

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/tagmanager/components/setup/SetupForm.js [hunk 1]
```diff
@@ -84,6 +85,21 @@ export default function SetupForm( { finishSetup } ) {
...
+	);
+	const isNavigating = useSelect( ( select ) =>
+		select( CORE_LOCATION ).isNavigating()
+	);
+	const isSaving = isDoingSubmitChanges || isNavigating || submitInProgress;
+
+	// This flag is be used to determine whether to show the loading spinner
+	// within the "Continue to Analytics setup" button when setting up GTM with Analytics.
+	// It prevents the spinner from showing when the user opts to set up GTM without Analytics.
+	const [ savingGA4AndGTM, setSavingGA4AndGTM ] = useState( false );
```
[the remark this answers is not in the document]
Comment (hussain-t) [reply assets/js/modules/tagmanager/components/setup/SetupForm.js]: Updated ✅

File: assets/js/modules/adsense/components/setup/SetupAccountSite/SetupAccountSiteUI.js [hunk 2]
```diff
@@ -29,17 +29,27 @@ import { Fragment } from '@wordpress/element';
...
 	description,
 	primaryButton,
 	secondaryButton,
 } ) {
+	const isDoingSubmitChanges = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).isDoingSubmitChanges()
+	);
+	const isNavigating = useSelect( ( select ) =>
+		select( CORE_LOCATION ).isNavigating()
+	);
```
[the remark this answers is not in the document]
Comment (hussain-t) [reply assets/js/modules/adsense/components/setup/SetupAccountSite/SetupAccountSiteUI.js]: Updated ✅

File: assets/js/modules/analytics/components/setup/SetupForm.js [hunk 3]
```diff
@@ -74,6 +75,12 @@ export default function SetupForm( { finishSetup } ) {
 
 		return canSubmitChanges;
 	} );
+	const isDoingSubmitChanges = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS ).isDoingSubmitChanges()
+	);
+	const isNavigating = useSelect( ( select ) =>
+		select( CORE_LOCATION ).isNavigating()
+	);
```
[the remark this answers is not in the document]
Comment (hussain-t) [reply assets/js/modules/analytics/components/setup/SetupForm.js]: Updated ✅

File: assets/js/modules/tagmanager/components/setup/SetupForm.js [hunk 4]
```diff
@@ -84,6 +85,21 @@ export default function SetupForm( { finishSetup } ) {
...
+	const isDoingSubmitChanges = useSelect( ( select ) =>
+		select( MODULES_TAGMANAGER ).isDoingSubmitChanges()
+	);
+	const submitInProgress = useSelect( ( select ) =>
+		select( CORE_FORMS ).getValue( FORM_SETUP, 'submitInProgress' )
+	);
+	const isNavigating = useSelect( ( select ) =>
+		select( CORE_LOCATION ).isNavigating()
+	);
+	const isSaving = isDoingSubmitChanges || isNavigating || submitInProgress;
```
[the remark this answers is not in the document]
Comment (hussain-t) [reply assets/js/modules/tagmanager/components/setup/SetupForm.js]: Updated ✅

**Review by techanvil — APPROVED** [review]
LGTM, nice one @hussain-t!

**Pull request merged**